arrow-core-data/src/main/kotlin/arrow/core/Either.kt

```diff
@@ -1041,24 +1162,52 @@ sealed class Either<out A, out B> : EitherOf<A, B> {
     inline fun <L, R> conditionally(test: Boolean, ifFalse: () -> L, ifTrue: () -> R): Either<L, R> =
       if (test) right(ifTrue()) else left(ifFalse())
 
+    @Deprecated("Use the inline version. Hidden for binary compat", level = DeprecationLevel.HIDDEN)
     suspend inline fun <R> catch(f: suspend () -> R): Either<Throwable, R> =
+      catch { f() }
+
+    inline fun <R> catch(f: () -> R): Either<Throwable, R> =
```

Comment:
This breaks the API in some places where you pass a `suspend` lambda directly to the `catch` function, I think the Kotlin compiler has the same limitation for suspend method references..

See https://github.com/arrow-kt/arrow-fx/pull/354 as example of this breakage.